lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -421,6 +421,24 @@ void tools::AddGoldPlugin(const ToolChain &ToolChain, const ArgList &Args,
       CmdArgs.push_back(
           Args.MakeArgString(Twine("-plugin-opt=sample-profile=") + FName));
   }
+
+  // Support for struct field cache analysis
```

Comment:
As we discussed yesterday, you don't need to do anything special to access profile data. If you pass  -fprofile-use, then there is a pass that annotates the function entry count and the branch probabilities. You should then be able to use BlockFrequencyInfo to get the count of a given block.